python/ray/data/tests/test_numpy_support.py

```diff
@@ -22,8 +22,8 @@ def do_map_batches(data):
 
 
 def assert_structure_equals(a, b):
-    assert type(a) == type(b), (type(a), type(b))
-    assert type(a[0]) == type(b[0]), (type(a[0]), type(b[0]))  # noqa: E721
+    assert isinstance(a, type(b)), (type(a), type(b))
+    assert isinstance(a[0], type(b[0])), (type(a[0]), type(b[0]))
```

Comment:
```python
assert type(a) is type(b), (type(a), type(b))
assert type(a[0]) is type(b[0]), (type(a[0]), type(b[0]))
```

Reply:
Thanks for the feedback! I've made the changes you suggested. Let me know if anything else needs tweaking.